import-dashboards.py

```diff
@@ -529,9 +530,84 @@ def set_home_dashboard(api_key):
     #             "SELECT 1, 1, 0, 0, id, '', '', datetime('now'), datetime('now') from dashboard WHERE slug='home'")
 
 
+def dbaas_dashboard(api_key):
+    if DBAAS is None or DBAAS.lower() in ('0','false','f'):
+        print ' * DBaaS is disabled'
+        r = requests.get('%s/api/dashboards/uid/pmm-dbaas' % (HOST,), headers=grafana_headers(api_key))
+        if r.status_code == httplib.OK:
+            print '   * Removing DBaaS dashboard'
+            r = requests.delete('%s/api/dashboards/uid/pmm-dbaas' % (HOST,), headers=grafana_headers(api_key))
+            if r.status_code != httplib.OK:
+                print r.status_code, r.content
+        return
+
+    print ' * DBaaS is enabled'
+    r = requests.get('%s/api/dashboards/uid/pmm-dbaas' % (HOST,), headers=grafana_headers(api_key))
+    if r.status_code != httplib.OK:
+        print '   * Adding DBaaS dashboard'
+        with open('/usr/share/percona-dashboards/pmm-app/dist/dashboards/pmm-dbaas.json', 'r') as dashboard_file:
+            dashboard = json.loads(dashboard_file.read())
+        data = json.dumps({'dashboard': dashboard})
+        r = requests.post('%s/api/dashboards/import' % (HOST,), data=data, headers=grafana_headers(api_key))
+        if r.status_code != httplib.OK:
+            print r.status_code, r.content
+    else:
+        print '   * DBaaS dashboard has already been added'
+
+
+def check_active_process(processname):
+    ps = os.popen("ps -Af").read()
+    proccount = ps.count(processname)
+
+    if proccount > 0:
+        print' * %s service is running' % (processname,)
+        return True
+
+
+def args_parser():
+   parser = argparse.ArgumentParser()
+   parser.add_argument("grafana_db_dir", type=str, nargs='?', default="/srv/grafana", help="Set Grafana DB folder")
+   parser.add_argument("-d", "--dbaas", required=False, help="Perform only operation of add/remove dbass dashboard", action="store_true")
+   args = parser.parse_args()
+   return (args.dbaas, args.grafana_db_dir)
+
+
 def main():
-    print "Grafana database directory: %s" % (GRAFANA_DB_DIR,)
-    upgrade = check_dashboards_version()
+    # Add/Remove DBaaS dashboard PMM-7085
+    is_dbaas, grafana_db_dir = args_parser()
+    if is_dbaas:
```

Comment:
can we move everything under this `if` except `return` to a separate method and add TODO to delete it once DBaaS will be GA?

Reply:
Separate method _dbaas_dashboard_ has already been created. It's also used during updates.
All other commands in this sections are requests to methods for getting/removing an api_key.
And such methods are called for other script actions.

IMHO it would be more complicate to figure out what dependencies are related to a new general method only. 
So I propose to left it in a section with the TODO comment.

Reply:
let's add comment in all places where it should be removed

Reply:
Done